### PR TITLE
HTTP: move utils to a proper file, add tests

### PR DESCRIPTION
**Pull request by virtualtam**
Relates to #333

Modifications:
 - move HTTP utils to 'application/HttpUtils.php'
 - simplify logic
   - replace 'http_parse_headers_shaarli' by built-in 'get_headers()'
   - remove superfluous '$status' parameter (provided by the HTTP headers)
 - apply coding conventions
 - add test coverage (unitary only)

### CODE REVIEW

File: application/HttpUtils.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ */
+function get_http_url($url, $timeout = 30)
+{
+    $options = array(
+        'http' => array(
+            'method' => 'GET',
+            'timeout' => $timeout,
+            'user_agent' => 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:23.0)'
+                         .' Gecko/20100101 Firefox/23.0'
+        )
```
Comment (ArthurHoaro) [inline application/HttpUtils.php]: Specifying Shaarli in the UA would be nice.

**Comment by ArthurHoaro** [comment]
Headers are only used to check if the response code is 200. I think that the code could be simplified with `get_http_url` return the content if the HTTP code is 200, false otherwise.

We can add more featured functions later if the need comes.

File: application/HttpUtils.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    );
+    $context = stream_context_create($options);
+
+    // limit downloaded data to 4 MiB (4,194,304 bytes)
```
Comment (ArthurHoaro) [inline application/HttpUtils.php]: Maybe this should be a parameter.

File: application/HttpUtils.php (hunk 1 above)
Comment (virtualtam) [inline application/HttpUtils.php]: unless there are download issues due to an obsolete UA, I'd prefer keeping a standard value here

File: application/HttpUtils.php (hunk 2 above)
Comment (virtualtam) [inline application/HttpUtils.php]: +1

the UA could also be set as a parameter

**Comment by virtualtam** [comment]
+1 for simplifying the method further

I'd prefer addressing it when refactoring the thumbnail download / generation, because this is where the duplication & complexity happens

**Comment by virtualtam** [comment]
#345 raised to refactor calling code (thumbnail generation) and simplify `get_http_url`

**Pull request merged**